subprojects/core/src/main/java/org/gradle/invocation/DefaultGradle.java

```diff
@@ -92,9 +91,9 @@ public abstract class DefaultGradle extends AbstractPluginAware implements Gradl
     private final GradleLifecycleActionExecutor gradleLifecycleActionExecutor;
     private List<IncludedBuildInternal> includedBuilds;
     private final MutableActionSet<Project> rootProjectActions = new MutableActionSet<>();
+    private final MutableActionSet<GradleInternal> baseProjectClassLoaderLockedActions = new MutableActionSet<>();
     private final IsolatedProjectEvaluationListenerProvider isolatedProjectEvaluationListenerProvider;
     private GradleLifecycle lifecycle;
-    private boolean projectsLoaded;
```

Comment:
ℹ️ Don't need any extra state. The state we're looking for is already beared by baseProjectClassLoaderScope